### ISSUE TITLE
Upgrade yew

### ISSUE THREAD

**Issue by ctm** [ctm/mb2-doc#757]
Upgrade yew from 0.18.0 to 0.19.3 and yew-router from 0.15.0 to 0.16.0.

I haven't been paying _close_ attention to what's happening with yew, so I don't know how big of a task this is, but I did do a few things in anticipation already, so perhaps this won't be too bad.

**Comment by ctm** [comment]
It took me more than a day, although to be fair, I chose to have a couple beers yesterday in solidarity with the folks at the social events I skipped. The diff was 4,501 lines long and it _almost_ ran on the first try. The first try brought up the lobby, but the tables didn't come up because I hadn't paid enough attention when I rewrote the router, so to get the tables working I needed to make these changes:
```
diff --git a/front-ends/web/src/lib.rs b/front-ends/web/src/lib.rs
index b5b23a9a..85d72a75 100644
--- a/front-ends/web/src/lib.rs
+++ b/front-ends/web/src/lib.rs
@@ -42,14 +42,14 @@ use {
 
 #[derive(Clone, Debug, Routable, PartialEq)]
 pub enum Route {
-    #[at("/event/{event_id}/player/{player_id}")]
+    #[at("/event/:event_id/player/:player_id")]
     EventPlayer {
         event_id: EventId,
         player_id: PlayerId,
     },
-    #[at("/event/{id}")]
+    #[at("/event/:id")]
     Event { id: EventId },
-    #[at("/table/{id}")]
+    #[at("/table/:id")]
     Table { id: TableId },
     #[at("/chat")]
     Chat,
```
In other words, there was a syntax change in the DSL used to specify routes and nothing noticed that I was using the old syntax. This is not a complaint, just an observation. I'm super happy that Rust allows nice refactorings like this.

FWIW, I did a little bit of cleanup here and there as I went. I know a lot more about Rust as well as the design patterns associated with Yew than I did when I started. OTOH, I was eager to push forward so I could deploy in time for this evening's tournament. I didn't know how long it would take, so I didn't go very deep into any refactoring rabbit-holes.

Whee!

**Comment by ctm** [comment]
Oops. Deployed too soon. Found a showstopper and had to rollback.

**Comment by ctm** [comment]
I fixed the showstopper, did a bunch more testing and redeployed.
